src/utils/routing.ts

```diff
@@ -0,0 +1,67 @@
+import { normalizePath } from './fs';
+
+/**
+ * Strip route groups from a path name.
+ *
+ * The build output config does not rewrite requests to route groups, so we need to strip route
+ * groups from the path name for file system matching.
+ *
+ * @param path Path name to strip route groups from.
+ * @returns Path name with route groups stripped.
+ */
+export function stripRouteGroups(path: string) {
+	return path.replace(/\/\(([^)]+)\)/g, '');
+}
+
+/**
+ * Strip `/index` from a path name.
+ *
+ * The build output config does not rewrite `/` to `/index`, so we need to strip `/index` from the
+ * path name for request matching.
+ *
+ * @param path Path name to strip `/index` from.
+ * @returns Path name with `/index` stripped.
+ */
+export function stripIndexRoute(path: string) {
+	// add leading slash back if it is stripped when `/index` is removed
+	return addLeadingSlash(path.replace(/\/index$/, ''));
```

Comment:
```suggestion
	return addLeadingSlash(path.replace(/^\/index$/, ''));
```

Reply:
spoke with @james-elicx, we might want to check that the `/index` is just at the end as well (or anyways it doesn't do any harm not checking that the string is precisely the `/index`)